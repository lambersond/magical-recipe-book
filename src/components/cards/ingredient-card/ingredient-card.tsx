import clsx from 'classnames'
import { BiomeChip, RarityChip } from '@/components/chips'
import { Card, Tooltip } from '@/components/common'
import { RARITY_LABELS } from '@/constants/rarity'
import { getBorderColorByRarity } from '@/utils/rarity'
import type { Ingredient } from '@/types'

export function IngredientCard({
  bane,
  biomes,
  boon,
  description,
  name,
  rarity,
}: Readonly<Ingredient>) {
  const cardClasses = clsx(
    'border-t-2 flex flex-col',
    getBorderColorByRarity(rarity),
  )

  return (
    <Card className={cardClasses}>
      <div className='flex items-center items-center justify-between mb-2'>
        <p className='text-xl font-semibold text-text-primary'>{name}</p>
        <Tooltip title={RARITY_LABELS[rarity]}>
          <RarityChip rarity={rarity} iconOnly />
        </Tooltip>
      </div>
      <div className='text-text-tertiary mb-4 font-thin italic text-sm'>
        {description}
      </div>
      <div className='flex flex-col space-y-1 text-sm'>
        {boon && (
          <span className='font-semibold text-text-tertiary'>
            Boon:{' '}
            <span className='text-text-primary font-normal text-text-secondary'>
              {boon}
            </span>
          </span>
        )}
        {bane && (
          <span className='font-semibold text-text-tertiary'>
            Bane:{' '}
            <span className='text-text-primary font-normal text-text-secondary'>
              {bane}
            </span>
          </span>
        )}
      </div>
      {biomes && (
        <div className='mt-4 text-sm text-text-tertiary flex gap-1 flex-wrap'>
          {biomes.map(({ biome }) => (
            <BiomeChip
              key={biome.id}
              name={biome.name}
              description={biome.description ?? ''}
            />
          ))}
        </div>
      )}
    </Card>
  )
}
