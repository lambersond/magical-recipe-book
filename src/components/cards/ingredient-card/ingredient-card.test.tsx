import { render } from '@test-utils'
import { IngredientCard } from './ingredient-card'
import type { Ingredient } from '@/types'

describe('components/cards/ingredient-card', () => {
  const ingredient = {
    id: 'abc',
    name: 'Name',
    description: '',
    bane: 'Bane',
    boon: 'Boon',
    rarity: 'common',
    biomes: [
      {
        biome: {
          id: '1',
          name: 'Biome 1',
          description: 'Description 1',
        },
      },
      {
        biome: {
          id: '2',
          name: 'Biome 2',
          description: 'Description 2',
        },
      },
    ],
    createdAt: new Date('2023-01-01T00:00:00Z'),
    updatedAt: new Date('2023-01-01T00:00:00Z'),
  } as any as Ingredient

  it('should match snapshot', () => {
    expect(
      render(<IngredientCard {...ingredient} />).asFragment(),
    ).toMatchSnapshot()
  })
})
