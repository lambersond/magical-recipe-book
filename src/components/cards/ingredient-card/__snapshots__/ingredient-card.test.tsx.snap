// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`components/cards/ingredient-card should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="p-4 border-border bg-card w-full rounded-none sm:rounded-lg border-t-2 flex flex-col border-rarity-common"
    data-testid="Card"
  >
    <div
      class="flex items-center items-center justify-between mb-2"
    >
      <p
        class="text-xl font-semibold text-text-primary"
      >
        Name
      </p>
      <button
        class="w-fit"
        data-state="closed"
      >
        <svg
          class="inline size-4 text-text-rarity-common"
          data-testid="D4Icon"
          height="24"
          preserveAspectRatio="xMidYMid meet"
          viewBox="0 0 5120 5120"
          width="24"
        >
          <path
            d="M2349 5024 c-25 -38 -113 -157 -196 -264 -154 -202 -518 -680 -1064 -1400 -174 -228 -450 -593 -615 -810 -389 -512 -474 -626 -474 -639 0 -22 466 -396 1216 -976 50 -38 246 -190 435 -337 189 -146 434 -336 544 -421 l200 -154 6 496 c8 647 8 4026 0 4335 l-6 239 -46 -69z M2710 5080 c-5 -40 -16 -4669 -12 -4833 l7 -227 265 203 c146 112 421 325 611 473 587 457 1227 955 1387 1078 83 65 152 120 152 123 0 9 -566 759 -1465 1943 -157 206 -428 564 -603 794 -367 486 -341 452 -342 446z"
            fill="currentColor"
          />
        </svg>
      </button>
    </div>
    <div
      class="text-text-tertiary mb-4 font-thin italic text-sm"
    />
    <div
      class="flex flex-col space-y-1 text-sm"
    >
      <span
        class="font-semibold text-text-tertiary"
      >
        Boon: 
        <span
          class="text-text-primary font-normal text-text-secondary"
        >
          Boon
        </span>
      </span>
      <span
        class="font-semibold text-text-tertiary"
      >
        Bane: 
        <span
          class="text-text-primary font-normal text-text-secondary"
        >
          Bane
        </span>
      </span>
    </div>
    <div
      class="mt-4 text-sm text-text-tertiary flex gap-1 flex-wrap"
    >
      <div
        class="text-xs rounded-sm px-2 py-1 border flex min-w-fit items-center"
        data-state="closed"
      >
        <p>
          Biome 1
        </p>
      </div>
      <div
        class="text-xs rounded-sm px-2 py-1 border flex min-w-fit items-center"
        data-state="closed"
      >
        <p>
          Biome 2
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
